### PR TITLE
Create JaxrsModule.adminOnlyJaxrsModule()

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java [hunk 1]
```diff
@@ -271,48 +158,220 @@ else if (type.isAnnotationPresent(Path.class)) {
...
-            ServiceLocator locator = locatorReference.get();
-            locator.inject(object);
-            locator.postConstruct(object);
-            return object;
+            binder.disableCircularProxies();
+
+            jaxrsBinder(binder).bind(JsonMapper.class);
+            jaxrsBinder(binder).bind(SmileMapper.class);
+            jaxrsBinder(binder).bind(TimingResourceDynamicFeature.class);
+            jaxrsBinder(binder).bind(DisallowOptionsModelProcessor.class);
```
Comment (lamielle) [inline jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java]: If I read properly, this is only relevant to JAX-RS applications, so maybe it doesn't make sense in the common module?

Similar comment for TimingResourceDynamicFeature.

File: jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java [hunk 2]
```diff
@@ -271,48 +158,220 @@ else if (type.isAnnotationPresent(Path.class)) {
...
+            jaxrsBinder(binder).bindInjectionProvider(ClientInfo.class).to(ClientInfoSupplier.class);
+            jaxrsBinder(binder).bindAdmin(ParsingExceptionMapper.class);
+            jaxrsBinder(binder).bindAdmin(QueryParamExceptionMapper.class);
+            jaxrsBinder(binder).bindAdmin(OverrideMethodFilter.class);
+            jaxrsBinder(binder).bindAdmin(ThreadDumpResource.class);
+
+            newSetBinder(binder, Object.class, JaxrsResource.class).permitDuplicates();
```
Comment (lamielle) [inline jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java]: I don't think these three lines are needed for a non-JAX-RS application either.  At least, the module I implemented that does effectively what you're doing here did not need them.

File: jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java (hunk 1 above)
Comment (johngmyers) [inline jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java]: I thought it disallowed OPTIONS on the admin port as well, but apparently not.

**Comment by lamielle** [comment]
I pulled your branch and after a brief test it seems using adminOnlyJaxrsModule is working as expected!

File: jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java (hunk 2 above)
Comment (johngmyers) [inline jaxrs/src/main/java/com/proofpoint/jaxrs/JaxrsModule.java]: Need these to create the Application. I believe non-`Resource` singletons such as the `InjectionProvider`s also apply to the admin port.

**Comment by johngmyers** [comment]
Squashed and merged

**Pull request closed without merging**